### PR TITLE
Windows XP support

### PR DESCRIPTION
**Pull request by jha334201553**
1. 兼容 xp 的 EPROCESS struct
2. 兼容 windows 内核 在 single cpu 的情况

### CODE REVIEW

File: src/libdrakvuf/win-offsets.h [hunk 1]
```diff
@@ -158,7 +158,8 @@ enum win_offsets
     KPCR_PRCBDATA,
     KPRCB_CURRENTTHREAD,
 
-    KTHREAD_PROCESS,
+    KTHREAD_PROCESS, 
+    ETHREAD_PROCESS_XP,  //  xp use this value
```
Comment (tklengyel) [inline src/libdrakvuf/win-offsets.h]: Please use `ETHREAD_THREADSPROCESS` as the enum name

File: src/libdrakvuf/drakvuf.c [hunk 2]
```diff
@@ -229,6 +229,7 @@ bool drakvuf_init(drakvuf_t* drakvuf, const char* domain, const char* rekall_pro
     return 1;
 
 err:
+    drakvuf_resume(*drakvuf);  // 失败的时候恢复暂停系统，否则只能强制关闭虚拟机
```
Comment (tklengyel) [inline src/libdrakvuf/drakvuf.c]: This should not be necessary.

File: src/libdrakvuf/win-processes.c [hunk 3]
```diff
@@ -200,11 +200,29 @@ addr_t win_get_current_thread(drakvuf_t drakvuf, drakvuf_trap_info_t* info)
...
+    addr_t process = 0;
 
     thread=win_get_current_thread(drakvuf, info);
 
-    if (thread == 0 || VMI_SUCCESS != vmi_read_addr_va(drakvuf->vmi, thread + drakvuf->offsets[KTHREAD_PROCESS], 0, &process))
+    if (thread == 0)
+    {
+        return 0;
+    }
+    if (drakvuf->offsets[KTHREAD_PROCESS] != 0)
```
Comment (tklengyel) [inline src/libdrakvuf/win-processes.c]: LibVMI should be able to return the Windows version, so this check should be explicitely whether we have an XP guest or not.

**Comment by tklengyel** [comment]
Thanks for this PR, XP support has been on the todo for a long time now. Could you also say whether injection works against an XP guest? If not, please also add a check there that quits early.

**Comment by tklengyel** [comment]
@drakvuf-jenkins This is OK to test

**Comment by tklengyel** [comment]
@jha334201553 are you still working on this PR?

**Comment by jha334201553** [comment]
> @ jha334201553您是否还在从事此PR？

Sorry, I quit

**Comment by tklengyel** [comment]
OK, thanks for the update

**Pull request closed without merging**